include/picongpu/particles/collision/detail/ListEntry.hpp

```diff
@@ -87,24 +47,58 @@ namespace picongpu::particles::collision
             //! number of particles per cell
             memory::Array<uint32_t, T_numElem> numParticles;
 
-            /** Frame pointer array.
+            using FramePtrType = typename T_ParticlesBox::FramePtr;
+
+            /**! pointer to an array of frames for the selected supercell index
              *
-             * A Frame pointer contains only a pointer therefore storing data as void* is allowed (but not
-             * nice). This keeps the ListEntry signature equal for all species.
+             * Since ListEntry is designed to be located in shared memory the list located in global memory will be
+             * shared by all workers.
              */
-            void** framePtr = nullptr;
+            FramePtrType* framePtr = nullptr;
 
         public:
-            DINLINE uint32_t& size(uint32_t cellIdx)
+            /** Access all particles in a cell.
+             *
+             * An accessor provides access to all particles within a cell (NOT supercell).
+             *
+             */
+
+            struct ParticleAccessor
             {
-                return numParticles[cellIdx];
-            }
+                FramePtrType* m_framePtrList = nullptr;
+                uint32_t* m_parIdxList = nullptr;
+                uint32_t m_numPars = 0u;
+
+                static constexpr uint32_t frameSize = FramePtrType::type::frameSize;
+
+                DINLINE ParticleAccessor(uint32_t* parIdxList, uint32_t const numParticles, FramePtrType* framePtrList)
+                    : m_framePtrList(framePtrList)
+                    , m_parIdxList(parIdxList)
+                    , m_numPars(numParticles)
+                {
+                }
+
+                /** Number of particles within the cell
+                 */
+                DINLINE uint32_t size() const
+                {
+                    return m_numPars;
+                }
+
+                /** get particle
+                 *
+                 * @param idx index of the particle, range [0;size())
+                 */
+                DINLINE auto operator[](uint32_t idx) const
+                {
+                    const uint32_t inSuperCellIdx = m_parIdxList[idx];
+                    return m_framePtrList[inSuperCellIdx / frameSize][inSuperCellIdx % frameSize];
```

Comment:
With both integers? I don't think so

Reply:
Both types a integral types, they can not end in floating point numbers, that are defined by the language.
A comment is not useful because it will increase the code size without extra new knowlage.